core-backend/common/src/lib.rs

```diff
@@ -110,25 +109,20 @@ pub struct ExtInfo {
 }
 
 pub trait IntoExtInfo {
-    fn into_ext_info(
-        self,
-        memory: &impl Memory,
-    ) -> Result<(ExtInfo, Option<TrapExplanation>), (MemoryError, GasAmount)>;
+    fn into_ext_info(self, memory: &impl Memory) -> Result<ExtInfo, (MemoryError, GasAmount)>;
 
     fn into_gas_amount(self) -> GasAmount;
 
     fn last_error(&self) -> Option<&ExtError>;
-}
 
-pub struct BackendReport {
-    pub termination: TerminationReason,
-    pub info: ExtInfo,
+    fn trap_explanation(&self) -> Option<TrapExplanation>;
 }
 
+pub type BackendReport<T> = (TerminationReason, T, Option<WasmPageNumber>);
+
 #[derive(Debug, derive_more::Display)]
 #[display(fmt = "{}", reason)]
 pub struct BackendError<T> {
-    pub gas_amount: GasAmount,
```

Comment:
Why the field was removed?

Reply:
We don't want to destroy ext inside backend.